### PR TITLE
sprint challenge submission

### PR DESCRIPTION
**Pull request by sharp0111**

### CODE REVIEW

**Comment by sharp0111** [comment]
couldn't finish the code, but had a strong feeling in the beginning that I could handle that :) Oh well...

**Comment by sharp0111** [comment]
please, check and tell me if I'm getting close. Please!

File: ANSWERS.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Questions
+1. What is the DOM?
+- (DOM) stands for document object model and is a programming interface for HTML and XML documents. It represents the page so that programs can change the document structure, style, and content. The DOM represents the document as nodes and objects. That way, programming languages can connect to the page.
+2. What is an event?
+- Event interface represents any event which takes place in the DOM; some are user-generated (such as mouse or keyboard events), while others are generated by APIs (such as events that indicate an animation has finished running, a video has been paused, and so forth). There are many types of events, some of which use other interfaces based on the main Event interface. Event itself contains the properties and methods which are common to all events.
```
Comment (Velsu) [inline ANSWERS.md:5]: Great answers.

File: components/Carousel/Carousel.js [hunk 2]
```diff
@@ -1,6 +1,55 @@
...
+            }
+        });
+
+        this.activeData = this.element.querySelector('.Carousel__item-focused');
+        this.activeData = this.activeData ? this.activeData.dataset.num : null;
+        this.updateActive(this.activeData);
+    }
 
+    updateActive(data) {
+        if (data === null) return;
```
Comment (Velsu) [inline components/Carousel/Carousel.js:46]: The Carousel code was implemented in interesting manner but it is not working. I think You wanted to aim high and write very clean and modular code which is pro, but if You want to make this code work there is some less clean but easier way to do it.

I would split the `updateActive` to separate functions for left and right arrow. Then catch both of these into variables and start operating on that, make an Array of the content and just increment and decrement to display the text while listening for events on both arrows.

Also for current code You could also try to catch if arrow is left or right with this `event.target.getAttribute('data-arrow')` which will return either left or right.

There are lots of ways to do it and it is sometimes best to just get it working, even if code looks messy. Anyway very good job.

**Review by Velsu — COMMENTED** [review]
Great job on this sprint challenge. I can see there was a lot of effort put into it. You aimed for clean and modular code but that backfired. For future i would advise to start with more straightforward way to solve the issue and then try to improve on that.

Anyway good job 👍

**Pull request closed without merging**